src/components/flow/routers/webhook/WebhookRouterForm.tsx

```diff
@@ -236,49 +236,47 @@ export default class WebhookRouterForm extends React.Component<
 
     const method = this.state.method.value.value;
     const name = this.state.method.value.label + ' ' + i18n.t('body', 'Body');
-    if (method === Methods.POST || method === Methods.PUT) {
-      tabs.push({
-        name,
-        body: (
-          <div key="post_body" className={styles.body_form}>
-            <h4>{name}</h4>
-            <p>
+    tabs.push({
+      name,
+      body: (
+        <div key="post_body" className={styles.body_form}>
+          <h4>{name}</h4>
+          <p>
+            <Trans
+              i18nKey="forms.call_webhook.body_summary"
+              values={{ method: this.state.method.value.label }}
+            >
+              Modify the body of the [[method]] request that will be sent to your webhook.
+            </Trans>
+          </p>
+          <TextInputElement
+            __className={styles.req_body}
+            name={name}
+            showLabel={false}
+            entry={this.state.body}
+            onChange={this.handleBodyUpdate}
+            helpText={
               <Trans
                 i18nKey="forms.call_webhook.body_summary"
                 values={{ method: this.state.method.value.label }}
               >
                 Modify the body of the [[method]] request that will be sent to your webhook.
               </Trans>
-            </p>
-            <TextInputElement
-              __className={styles.req_body}
-              name={name}
-              showLabel={false}
-              entry={this.state.postBody}
-              onChange={this.handlePostBodyUpdate}
-              helpText={
-                <Trans
-                  i18nKey="forms.call_webhook.body_summary"
-                  values={{ method: this.state.method.value.label }}
-                >
-                  Modify the body of the [[method]] request that will be sent to your webhook.
-                </Trans>
-              }
-              onFieldFailures={(persistantFailures: ValidationFailure[]) => {
-                const postBody = { ...this.state.postBody, persistantFailures };
-                this.setState({
-                  postBody,
-                  valid: this.state.valid && !hasErrors(postBody)
-                });
-              }}
-              autocomplete={true}
-              textarea={true}
-            />
-          </div>
-        ),
-        checked: this.state.postBody.value !== DEFAULT_BODY
-      });
-    }
+            }
+            onFieldFailures={(persistantFailures: ValidationFailure[]) => {
+              const body = { ...this.state.body, persistantFailures };
+              this.setState({
+                body,
+                valid: this.state.valid && !hasErrors(body)
+              });
+            }}
+            autocomplete={true}
+            textarea={true}
+          />
+        </div>
+      ),
+      checked: this.state.body.value !== getDefaultBody(method)
```

Comment:
This will resolve to true if you toggle to post and then back to GET without modifying the body (which should be unchecked)